josh-proxy/src/bin/josh-proxy.rs

```diff
@@ -565,35 +792,32 @@ async fn call_service(
         headref = "HEAD".to_string();
     }
 
-    if !josh_proxy::auth::check_auth(
-        &remote_url,
-        &auth,
-        ARGS.get_flag("require-auth") && parsed_url.pathinfo == "/git-receive-pack",
-    )
-    .in_current_span()
-    .await?
+    let http_auth_required =
+        ARGS.get_flag("require-auth") && parsed_url.pathinfo == "/git-receive-pack";
+
+    if !josh_proxy::auth::check_auth(&remote_url, &auth, http_auth_required)
+        .in_current_span()
+        .await?
     {
         tracing::trace!("require-auth");
         let builder = Response::builder()
-            .header("WWW-Authenticate", "Basic realm=User Visible Realm")
+            .header(
+                reqwest::header::WWW_AUTHENTICATE,
```

Comment:
reqwest is a crate dependency that I'd like to get rid of, so I'd prefer not introducing additional usage of it's symbols.